hai/parallel.py

```diff
@@ -238,3 +216,59 @@ def exceptions(self) -> Dict[str, Exception]:
             for t in self.tasks
             if t.ready() and not t._success  # type: ignore[attr-defined]
         }
+
+
+class ChordParallelRun(BaseParallelRun):
+    """Parallel run that inherits the thread pool from another parallel run
+    instead of managing its own
+    """
+
+    def __init__(self, main_run: "ParallelRun") -> None:
+        super().__init__()
+        self.main_run = main_run
+        self.pool = main_run.pool
+
+    def ready(self) -> bool:
+        return all([t.ready() for t in self.tasks])
+
+
+class ParallelRun(BaseParallelRun):
+    """
+    Encapsulates running several functions in parallel.
+
+    Due to the GIL, this is mainly useful for IO-bound threads, such as
+    downloading stuff from the Internet, or writing big buffers of data.
+
+    It's recommended to use this in a `with` block, to ensure the thread pool
+    gets properly shut down.
+    """
+
+    def __init__(self, parallelism: Optional[int] = None) -> None:
+        super().__init__()
+        self.pool = ThreadPool(
+            processes=(parallelism or (int(os.cpu_count() or 1) * 2)),
+        )
+        self.chords: WeakSet[ChordParallelRun] = WeakSet()
```

Comment:
What would one use this field for?

Reply:
It supports a pattern where tasks can create chords for the main parallel run thread to wait on. It could be something users could implement themselves though.